bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryClient.java

```diff
@@ -122,15 +122,16 @@ public static synchronized void runCleanupJobs() {
    *
    * @param job The {@code Job} to keep track of.
    */
-  public JobInfo waitForJob(Job job) {
+  public JobInfo waitForJob(Job job, long jobTimeoutInMinutes) {
     try {
       Job completedJob =
           job.waitFor(
               RetryOption.initialRetryDelay(Duration.ofSeconds(1)),
-              RetryOption.totalTimeout(Duration.ofMinutes(3)));
-      if (completedJob == null && completedJob.getStatus().getError() != null) {
+              RetryOption.totalTimeout(Duration.ofMinutes(jobTimeoutInMinutes)));
+      if (completedJob == null || completedJob.getStatus().getError() != null) {
         throw new UncheckedIOException(
-            new IOException(completedJob.getStatus().getError().toString()));
+            new IOException(
+                completedJob != null ? completedJob.getStatus().getError().toString() : null));
```

Comment:
When do we get the job as null? 
Is it on timeout? If it is, we should add the appropriate message instead.

Reply:
`waitFor()` is a bigQuery API to get the job information which can return null.
We are throwing NPE here if the output is null.